flexport/latitude
# LDS-347 Column pinning is broken in IE

## Description:
Adds support for sticky header and sticky columns in Latitude Table in IE. No polyfill, no browser sniffing, implementation only applies to browsers that don't support `sticky` (https://caniuse.com/#search=position%3A%20sticky).

Note: May run into stuttering (on paint, no scroll stutter) on slower machines. This relies on the scroll event and transforms to recreate the sticky effect. It does not force rerenders or reflows but is bounded by scroll listener and browser paint performance  (uses `requestAnimationFrame`).

![11b9d7f7-3553-4090-8291-f11d9ae0d831](https://user-images.githubusercontent.com/7979583/67725570-15f56c80-f9a0-11e9-8706-94086ca03344.png)

## Sandbox server:
https://core-47.sandbox2.eng-dev.flexport.com/

## Jira:
[LDS-347](https://flexport.atlassian.net/browse/LDS-347)

## Manual Testing Plan:
1. Load Storybook in IE11 (BrowsterStack).
2. View all table stories.
3. Confirm tables with sticky columns and headers work correctly.

## Screenshots:
Old:
![table old](https://user-images.githubusercontent.com/7979583/67790066-56052f80-fa32-11e9-83f2-07ee0c3cceab.gif)

New:
![table new](https://user-images.githubusercontent.com/7979583/67789621-83051280-fa31-11e9-824b-e0d8aff95d7a.gif)
